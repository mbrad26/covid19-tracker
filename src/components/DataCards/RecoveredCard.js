import React from 'react';
import { Card } from 'react-bootstrap';
import CountUp from 'react-countup';

import ChartComponent from '../Chart/ChartComponent.js';

const RecoveredCard = ({
  country,
  recovered,
  lastUpdated,
  historicalData,
  recoveredIncrease,
  }) => {
  const labels = historicalData && Object.keys(historicalData);
  const values = historicalData && Object.values(historicalData);

  return (
    <Card className='shadow'>
      <Card.Body>
        <Card.Title>{country}: <span className='text-success font-weight-bold'>
          <CountUp
            prefix='+'
            start={0}
            end={recoveredIncrease}
            duration={2.5}
            separator=','
          />
          </span>
        </Card.Title>
        <ChartComponent
          labels={labels}
          values={values}
          label={'Recovered'}
          color={'#228B22'}
        />
        <Card.Text>
          {recovered.toLocaleString()}
          <span id='update'>{lastUpdated}</span>
        </Card.Text>
      </Card.Body>
    </Card>
  );
};

export default RecoveredCard;
