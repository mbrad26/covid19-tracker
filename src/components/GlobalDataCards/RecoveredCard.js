import React from 'react';
import { Card } from 'react-bootstrap';

const RecoveredCard = ({ data }) => (
  <Card className='shadow' id='success'>
    <Card.Body>
      <Card.Title>Recovered</Card.Title>
      <Card.Text>
        {data.toLocaleString()}
      </Card.Text>
    </Card.Body>
  </Card>
)

export default RecoveredCard;
