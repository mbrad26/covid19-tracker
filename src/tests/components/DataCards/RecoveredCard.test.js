import React from 'react';
// import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import { globalData, historicalData } from '../../fixtures';
import RecoveredCard from '../../../components/DataCards/RecoveredCard';

describe('RecoveredCard', () => {
  it('renderes snapshot', () => {
    // const { container } = render(
    //   <RecoveredCard
    //     recovered={globalData.recovered}
    //     country={globalData.country ? globalData.country : 'Worldwide'}
    //     lastUpdated={new Date(globalData.updated).toDateString()}
    //     recoveredIncrease={globalData.todayRecovered}
    //     historicalData={historicalData}
    //   />
    // );
    //
    // expect(container.firstChild).toMatchSnapshot();
  });
});
