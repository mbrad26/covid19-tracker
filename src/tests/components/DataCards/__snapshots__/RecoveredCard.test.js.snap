// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecoveredCard renderes snapshot 1`] = `
<div
  class="card"
>
  <div
    class="card-body"
  >
    <div
      class="card-title h5"
    >
      Worldwide
      : 
      <span>
        <span>
          +0
        </span>
      </span>
      <i
        class="fas fa-smile"
      />
    </div>
    <canvas
      class="chartjs-render-monitor"
      height="0"
      style="display: block;"
      width="0"
    />
    <p
      class="card-text"
    >
      17,994,215
      <span
        class="text-muted"
        id="update"
      >
        Tue Sep 01 2020
      </span>
    </p>
  </div>
</div>
`;
